torch/fx/passes/tensorify_python_scalars.py

```diff
@@ -0,0 +1,262 @@
+def _tensorify_python_scalars(
+    gm: GraphModule, shape_env: ShapeEnv, fake_mode: fake_tensor.FakeTensorMode
+) -> None:
+    """
+    Converts Python scalar operations into Tensor operations within the graph. This pass looks for
+    Tensor operations that involve SymFloat arguments and transforms them into equivalent operations
+    that use only Tensor inputs.
+
+    Args:
+        gm: The FX graph module representing the computation graph.
+        shape_env: The shape environment responsible for symbolic shape tracking and propagation
+        during graph transformations.
+
+    Returns:
+        None
+    """
+    import sympy
+
+    if not torch._functorch.config.tensorify_python_scalars:
+        return None
+
+    graph = gm.graph
+    tracer = fx.proxy.GraphAppendingTracer(graph)
+    expr_to_sym_proxy: dict[sympy.Expr, MetaProxy] = {}
+    expr_to_tensor_proxy: dict[sympy.Expr, MetaProxy] = {}
+
+    first_non_placeholder = None
+    placeholders = set()
+    for node in graph.nodes:
+        if node.op != "placeholder":
+            first_none_placeholder = node
+            break
+        else:
+            placeholders.add(node)
+
+    Analysis = TensorReferenceAnalysis
+
+    def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+        # sympy_interp() with hash consing, and special handling for
+        # generating constants correctly
+        from sympy import Integer, Number, Symbol
+        from sympy.logic.boolalg import BooleanAtom
+
+        from torch.utils._sympy.interp import _run_sympy_handler, sympy_interp
+
+        # hash cons
+        if isinstance(expr, Symbol) and expr not in expr_to_tensor_proxy:
+            # This is guaranteed to be populated by invariant established by
+            # insert_deferred_runtime_asserts
+            node = graph.call_function(
+                torch.ops.aten.scalar_tensor.default,
+                (expr_to_sym_proxy[expr].node,),
+            )
+            node.meta["val"] = torch.ops.aten.scalar_tensor.default(
+                expr_to_sym_proxy[expr].node.meta["val"]
+            )
+            expr_to_tensor_proxy[expr] = MetaProxy(
+                node,
+                tracer=tracer,
+            )
+
+        # cache constants, why not
+        if isinstance(expr, (Integer, Number, BooleanAtom)):
+            dtype = None
+            c: Union[bool, int, float]
+            if isinstance(expr, BooleanAtom):
+                dtype = torch.bool
+                c = bool(expr)
+            elif isinstance(expr, sympy.Integer):
+                dtype = torch.int64
+                c = int(expr)
+            elif isinstance(expr, sympy.Number):
+                dtype = torch.float64
+                c = float(expr)
+
+            node = graph.call_function(
+                torch.ops.aten.scalar_tensor.default, (c,), {"dtype": dtype}
+            )
+            node.meta["val"] = torch.ops.aten.scalar_tensor.default(c, dtype=dtype)
```

Comment:
I wonder if it makes sense to have a propogate_meta optional arg on graph.call_function

Reply:
After discussion w/ @Chillee and @zou3519, I'll likely instead refactor this (outside of this diff) to look like

```
graph.insert_traced_subgraph(
    torch.ops.aten.scalar_tensor.default, (c,), {"dtype": dtype}
)	
```

Where under the hood `insert_traced_subgraph` will call `make_fx`. This has the nicety that

1) Decompositions are handled out of the box
2) We also get meta prop out of the box